### PR TITLE
Add MPLS Support

### PR DESCRIPTION
**Pull request by vitalisator**
This introduce the generic MPLS support
- New database tables for MPLS LSPs and MPLS LSP Paths
- Actually discovery and polling on NOKIA SR OS (Timetra) devices
- New routing->MPLS HTML pages

_Global MPLS view_
image

_Device MPLS view_
image

Suggestions and improvements are welcome

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/defaults.inc.php [hunk 1]
```diff
@@ -487,6 +487,8 @@
 // Poller/Discovery Modules
 $config['enable_bgp'] = 1;
 // Enable BGP session collection and display
+$config['enable_mpls'] = 1;
```
Comment (murrant) [inline includes/defaults.inc.php]: I prefer to use the modules to enable/disable things.

**Comment by vitalisator** [comment]
@murrant, have not really idea how to do "new style" or even where to start.
I would give it a try if you give me some advice :-)

Can I just put my code inside public functions poll and discover? 
But I think there is more to do...

**Comment by murrant** [comment]
Basically, but you would be missing out if you don't use Eloquent to update the DB, it replaces 90% of the code...  You could convert it in steps.

Since you've added test data, I could take a crack some time.

**Comment by murrant** [comment]
@vitalisator ok, I updated this to the new style (and improved the infrastructure for that a bit)  Can you test it?

I did change the migration because vRtrMplsLspPathStatTable is not in the test data so, trying to save nulls to the DB for that data caused the queries to fail.  Perhaps more or less fields should be nullable?

**Pull request merged**